sql/src/main/java/io/crate/operation/collect/sources/ShardCollectSource.java

```diff
@@ -461,6 +462,10 @@ private CrateCollector getShardsCollector(RoutedCollectPhase collectPhase,
 
         for (Map.Entry<String, List<Integer>> indexShards : indexShardsMap.entrySet()) {
             String indexName = indexShards.getKey();
+            if(metaData.index(indexName) == null){
```

Comment:
We could also avoid the duplicate lookup by assigning the result of `metadata.index(indexName)` to a temporary variable and do the null check on that.

Also - using the indexName in the exception may be confusing. In the case of a partition this could be something cryptic. We could first check if it's a partition - and if so ignore the error and continue - otherwise raise it as it is here.